build.gradle

```diff
@@ -769,7 +769,8 @@ subprojects {
     scalaCompileOptions.additionalParameters += ["-Xfatal-warnings"]
 
     scalaCompileOptions.additionalParameters += ["-release", String.valueOf(minJavaVersion)]
-
+    sourceCompatibility = minJavaVersion
```

Comment:
Both fields are deprecated (https://docs.gradle.org/current/userguide/upgrading_version_8.html#javacompile_tasks_may_fail_when_using_a_jre_even_if_compilation_is_not_necessary), so maybe we can use options.compilerArgs += ["--release", String.valueOf(minJavaVersion)] (https://docs.gradle.org/current/dsl/org.gradle.api.tasks.scala.ScalaCompile.html#org.gradle.api.tasks.scala.ScalaCompile:options) instead?

By the way, #16965 aims to use a helper to align JVM options for both JavaCompile and ScalaCompile. The release setting can be unified through the helper as well.

Reply:
Thanks for the review. I have added ` options.compilerArgs += ["--release", String.valueOf(minJavaVersion)]` now